### ISSUE TITLE
Add provisional support for draft PEP 675 (LiteralString)

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#3023]
PEP 675 is pretty far along, and `LiteralString` was just added to `typing_extensions`.

[PEP 675](https://www.python.org/dev/peps/pep-0675/)

**Comment by decorator-factory** [comment]
One thing I miss in Python's type checking from `TypeScript` is inferring literal types. Example:
```py
type ParseResult<T> =
    | { ok: true, value: T, rest: string }
    | { ok: false, error: string }
type Parser<T> = (s: string) => ParseResult<T>


const parseLiteral = <S extends string>(lit: S): Parser<S> => source =>
    source.startsWith(lit)
    ? { ok: true, value: lit, rest: source.slice(lit.length) }
    : { ok: false, error: `Source doesn't start with: ${lit}` }


const parseComma = parseLiteral(",")
// parseComma: Parser<",">
```

It's not part of PEP 675, but would it be possible for a type variable bound to `LiteralString` to have this behaviour?

**Comment by erictraut** [comment]
This PEP hasn't even been approved yet, nor have any of the typeshed stubs been updated to accommodate `LiteralString`. It's a bit early to be talking about additional features that are built on top of this. :)

**Comment by erictraut** [comment]
This is included in pyright 1.1.221, which I just published. It will also be included in the next release of pylance.
